packages/gatsby/src/internal-plugins/functions/gatsby-node.ts

```diff
@@ -230,6 +232,23 @@ const createWebpackConfig = async ({
     return acc
   }, {} as Record<string, string>)
 
+  // Delete env variables commonly used in Node so we don't accidentally break
+  // libraries.
+  delete varsFromProcessEnv[`HOME`]
```

Comment:
Pulling in env variables wholesale can cause unintended side effects. So remove the common ones.